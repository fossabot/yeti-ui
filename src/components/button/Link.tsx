import * as React from 'react';

import {
  ILinkProp,
  listOfButtonProps,
  removeProps,
} from '../../utilities';
import { generateStyle } from './styles';

class Link extends React.PureComponent<ILinkProp, {}> {
  public render() {
    const styles = generateStyle(this.props);

    return (
      <a
        {...removeProps(this.props, listOfButtonProps)}
        {...styles}/>
    );
  }
}

export default Link;
