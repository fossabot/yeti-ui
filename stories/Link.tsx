import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, number, color } from '@storybook/addon-knobs';

import Link from '../src/components/button/Link'

const stories = storiesOf('Link', module);

const divWrapperStyle: object = {
  fontFamily: '"Helvetica Neue", Helvetica, Roboto, Arial, sans-serif',
  fontWeight: 'normal',
  lineHeight: '1.5',
  color: '#0a0a0a',
  MozOsxFontSmoothing: 'grayscale',
};

stories.addDecorator(withKnobs);

stories.add('Basics', () => {

  return (
    <div style={divWrapperStyle}>
      <Link href={'#'}>{text('Label1', 'Learn More')}</Link>&nbsp;
      <Link href={'#'}>{text('Label2', 'View All Features')}</Link>&nbsp;
      <Link href={'#'} buttonType={'success'}>{text('Label3', 'Save')}</Link>&nbsp;
      <Link href={'#'} buttonType={'alert'}>{text('Label4', 'Delete')}</Link>
    </div>
  );
});

stories.add('Sizing', () => {
  return (
    <div style={divWrapperStyle}>
      <Link href={'#'} buttonSize={'tiny'}>So Tiny</Link>&nbsp;
      <Link href={'#'} buttonSize={'small'}>So Small</Link>&nbsp;
      <Link href={'#'}>So Basic</Link>&nbsp;
      <Link href={'#'} buttonSize={'large'}>So Large</Link>&nbsp;
      <Link href={'#'} buttonSize={'expanded'}>Such Expand</Link>&nbsp;
      <Link href={'#'} buttonSize={'small'} isExpanded={true}>Wow, Small Expand</Link>
    </div>
  );
});

stories.add('Coloring', () => {
  return (
    <div style={divWrapperStyle}>
      <Link href={'#'} buttonType={'primary'}>Primary</Link>&nbsp;
      <Link href={'#'} buttonType={'secondary'}>Secondary</Link>&nbsp;
      <Link href={'#'} buttonType={'success'}>Success</Link>&nbsp;
      <Link href={'#'} buttonType={'alert'}>Alert</Link>&nbsp;
      <Link href={'#'} buttonType={'warning'}>Warning</Link>
    </div>
  );
});

stories.add('Hollow Style', () => {
  return (
    <div style={divWrapperStyle}>
      <Link href={'#'} isHollow={true}>Primary</Link>&nbsp;
      <Link href={'#'} isHollow={true} buttonType={'secondary'}>Secondary</Link>&nbsp;
      <Link href={'#'} isHollow={true} buttonType={'success'}>Success</Link>&nbsp;
      <Link href={'#'} isHollow={true} buttonType={'alert'}>Alert</Link>&nbsp;
      <Link href={'#'} isHollow={true} buttonType={'warning'}>Warning</Link>&nbsp;
      <Link href={'#'} isHollow={true} isDisabled={true}>Disabled</Link>
    </div>
  );
});

stories.add('Disabled', () => {
  return (
    <div style={divWrapperStyle}>
      <Link href={'#'} isDisabled={true} aria-disabled>Disabled</Link>&nbsp;
      <Link href={'#'} buttonType={'primary'} isDisabled={true}>Disabled</Link>&nbsp;
      <Link href={'#'} buttonType={'secondary'} isDisabled={true}>Disabled</Link>&nbsp;
      <Link href={'#'} buttonType={'success'} isDisabled={true}>Disabled</Link>&nbsp;
      <Link href={'#'} buttonType={'alert'} isDisabled={true}>Disabled</Link>&nbsp;
      <Link href={'#'} buttonType={'warning'} isDisabled={true}>Disabled</Link>&nbsp;
      <br/>
      <Link href={'#'} isHollow={true} isAriaDisabled={true} isDisabled={true}>Disabled</Link>&nbsp;
      <Link href={'#'} buttonType={'primary'} isHollow={true} isDisabled={true}>Disabled</Link>&nbsp;
      <Link href={'#'} buttonType={'secondary'} isHollow={true} isDisabled={true}>Disabled</Link>&nbsp;
      <Link href={'#'} buttonType={'success'} isHollow={true} isDisabled={true}>Disabled</Link>&nbsp;
      <Link href={'#'} buttonType={'alert'} isHollow={true} isDisabled={true}>Disabled</Link>&nbsp;
      <Link href={'#'} buttonType={'warning'} isHollow={true} isDisabled={true}>Disabled</Link>
    </div>

  );
});

stories.add('Clear Style', () => {
  return (
    <div style={divWrapperStyle}>
      <Link href={'#'} isClear={true}>Primary</Link>&nbsp;
      <Link href={'#'} isClear={true} buttonType={'secondary'}>Secondary</Link>&nbsp;
      <Link href={'#'} isClear={true} buttonType={'success'}>Success</Link>&nbsp;
      <Link href={'#'} isClear={true} buttonType={'alert'}>Alert</Link>&nbsp;
      <Link href={'#'} isClear={true} buttonType={'warning'}>Warning</Link>&nbsp;
      <Link href={'#'} isClear={true} isDisabled={true}>Disabled</Link>
    </div>
  );
});

stories.add('Dropdown Arrows', () => {
  return (
    <div style={divWrapperStyle}>
      <Link href={'#'} isDropdown={true} buttonSize={'tiny'}>Dropdown Link</Link>&nbsp;
      <Link href={'#'} isDropdown={true} buttonSize={'small'}>Dropdown Link</Link>&nbsp;
      <Link href={'#'} isDropdown={true}>Dropdown Link</Link>&nbsp;
      <Link href={'#'} isDropdown={true} buttonSize={'large'}>Dropdown Link</Link>&nbsp;
      <Link href={'#'} isDropdown={true} isExpanded={true}>Dropdown Link</Link>
    </div>
  );
});

stories.add('Accessibility', () => {
  return (
    <div style={divWrapperStyle}>
      <Link
        href={'#'}
        color={color('Color', undefined)}
        backgroundColor={color('Background Color', undefined)}
        size={text('Size', undefined)}
        isHollow={boolean('isHollow', false)}
        isExpanded={boolean('isExpanded', false)}
        isDisabled={boolean('isDisabled', false)}
        isDropdown={boolean('isDropdown', false)}
        onClick={action('clicked')}>{text('Label', 'Hello Link')}</Link>&nbsp;
      <Link href={'#'}>
        <span className={'show-for-sr'}>Close</span>
        <span aria-hidden={'true'}><i className={'fi-x'}/></span>
      </Link>
    </div>

  );
});
