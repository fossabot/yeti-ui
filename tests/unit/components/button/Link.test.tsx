import * as Chance from 'chance';
import { shallow } from 'enzyme';
import * as React from 'react';

import {
  getCssValueByProperty,
  resetHeadStyle,
} from '../../../utilities';

import {
  ButtonColors,
  ButtonSizes,
} from '../../../../src/enums';
import {
  FOUNDATION_PALETTE,
} from '../../../../src/styles/global';
import Link from '../../../../src/components/button/Link';

const chance = new Chance();

describe('<Link>', () => {

  beforeEach(() => {
    resetHeadStyle();
  });

  function mountPureComponent(props, elementText?: string) {
    const expectedText = elementText ? elementText : chance.string();
    const onClickHandler = jest.fn();
    const wrapper = shallow(
      <Link {...props} href="#">{expectedText}</Link>);

    return { wrapper, onClickHandler, expectedText };
  }

  test('should render the link', () => {
    const { wrapper } = mountPureComponent({}, 'Expected text');

    expect(wrapper.find('a')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  test('should set className to the link element', () => {
    const expectedClassName = chance.string();
    const { wrapper } = mountPureComponent({ className: expectedClassName });

    expect(wrapper.find('a').prop('className')).toMatch(expectedClassName);
  });

  test('should set the size of the link', () => {
    const { wrapper } = mountPureComponent({ buttonSize: 'small' });
    const currentValue = getCssValueByProperty('font-size');

    expect(currentValue).toMatch(ButtonSizes.SMALL);
    expect(wrapper.find('a').prop('buttonSize')).toBeUndefined();
  });

  test('should set the color of the link based on type', () => {
    const { wrapper } = mountPureComponent({ buttonType: ButtonColors.SUCCESS });
    const currentValue = getCssValueByProperty('background-color');

    expect(currentValue).toMatch(FOUNDATION_PALETTE.SUCCESS);
    expect(wrapper.find('a').prop('buttonType')).toBeUndefined();
  });

  test('should set the link to hollow', () => {
    const { wrapper } = mountPureComponent({ isHollow: true });
    const currentValue = getCssValueByProperty('background-color');

    expect(currentValue).toMatch('transparent');
    expect(wrapper.find('a').prop('isHollow')).toBeUndefined();
  });

  test('sets expanded', () => {
    const { wrapper } = mountPureComponent({ isExpanded: true });
    const currentValue = getCssValueByProperty('width');

    expect(currentValue).toMatch('100%');
    expect(wrapper.find('a').prop('isExpanded')).toBeUndefined();
  });

  test('sets disabled', () => {
    const { wrapper } = mountPureComponent({ isDisabled: true });
    const currentValue = getCssValueByProperty('cursor');

    expect(currentValue).toMatch('not-allowed');
    expect(wrapper.find('a').prop('isDisabled')).toBeUndefined();
  });

  test('sets dropdown', () => {
    const { wrapper } = mountPureComponent({ isDropdown: true });
    const currentValue = getCssValueByProperty('top');

    expect(currentValue).toMatch('0.4em');
    expect(wrapper.find('a').prop('isDropdown')).toBeUndefined();
  });

  // test('sets arrow only', () => {
  //     const component = render(<Button isArrowOnly/>);
  //     expect(component).to.have.className('arrow-only');
  //     expect(component).to.not.have.attr('isArrowOnly');
  // });

  test('sets contents', () => {
    const { wrapper, expectedText } = mountPureComponent({});

    expect(wrapper.find('a').text()).toBe(expectedText);
  });
});