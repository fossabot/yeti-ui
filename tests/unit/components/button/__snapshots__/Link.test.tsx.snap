// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link> should render the link 1`] = `
<a
  data-css-18toa6z=""
  href="#"
>
  Expected text
</a>
`;
